node_helper.js

```diff
@@ -22,6 +23,24 @@ module.exports = NodeHelper.create({
 
   // Override socketNotificationReceived method.
   socketNotificationReceived: function (notification, payload) {
+    if (notification === "AUTH_NEEDED") {
+      qr.toDataURL(payload.url, {
+        type: "image/png",
+        width: 320,
+        height: 320,
+        margin: 5,
+        errorCorrectionLevel: 'H',
+        quality: 0.95,
+      }, (err, qrUrl) => 
+      {
+        if (err) {
+          Log.warn(this.name + ": " + err);
+        }
+        payload.qrUrl = qrUrl;
+        this.sendSocketNotification("AUTH_NEEDED_QR", payload)
```

Comment:
Just to make sure, if there's an error then payload.qrUrl is assigned null/undefined? In case is not a falsey value I'd add a return value after the Log.warn just to be safe : ) 

```suggestion
        if (err) {
          Log.warn(this.name + ": " + err);
          return
        }
        payload.qrUrl = qrUrl;
        this.sendSocketNotification("AUTH_NEEDED_QR", payload)
```

Reply:
Thanks for pointing that! But we shouldn't return on that code because we need to send back the `AUTH_NEEDED_QR` notification, otherwise the calendar will be stuck with the "Loading..." screen.

I just committed a new change where, instead of calling return, I just set `payload.qrUrl` to `false`. With this, we should be fine I guess